### ISSUE TITLE
[Feature]: Update Braze for React Native 0.73 Support

### ISSUE THREAD

**Issue by AbhavThakur** [braze-inc/braze-react-native-sdk#240]
### What problem are you facing?

```
Execution failed for task ':braze_react-native-sdk:compileDebugKotlin'.
> 'compileDebugJavaWithJavac' task (current target is 17) and 'compileDebugKotlin' task (current target is 11) jvm 
target compatibility should be set to the same Java version.

```
### Workarounds

if we added java 17 in braze/build.gradle  and android/app/build.gradle file it
```
compileOptions {
    sourceCompatibility JavaVersion.VERSION_17
    targetCompatibility JavaVersion.VERSION_17
  }

  kotlinOptions {
    freeCompilerArgs = ['-Xjvm-default=all']
    jvmTarget = "17"
  }
```
#### showing error
`Execution failed for task ':app:checkUatDebugDuplicateClasses'.
> Could not resolve all files for configuration ':app:uatDebugRuntimeClasspath'. > Failed to transform gradle-api-8.3.jar (Gradle API) to match attributes {artifactType=enumerated-runtime-classes, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}. > Execution failed for JetifyTransform: /Users/abhavthakur/.gradle/caches/8.3/generated-gradle-jars/gradle-api-8.3.jar. > Failed to transform '/Users/abhavthakur/.gradle/caches/8.3/generated-gradle-jars/gradle-api-8.3.jar' using Jetifier. Reason: JDOMParseException, message: Error on line 76: The prefix "Xlint" for element "Xlint:all" is not bound.. (Run with --stacktrace for more details.) Suggestions: - Check out existing issues at https://issuetracker.google.com/issues?q=componentid:460323&s=modified_time:desc, it's possible that this issue has already been filed there. - If this issue has not been filed, please report it at https://issuetracker.google.com/issues/new?component=460323 (run with --stacktrace and provide a stack trace if possible). `

**Comment by AbhavThakur** [comment]
hi @jerielng  
Do we have a temporary fix for the issue, or we need to wait until it is resolved?

**Comment by yungblud** [comment]
https://github.com/braze-inc/braze-react-native-sdk/pull/241
I made a PR for this issue.

**Comment by AbhavThakur** [comment]
Thanks @yungblud  & @jerielng

**Comment by Matthew-Nguyen98** [comment]
Facing the same issue +1

**Comment by thainx** [comment]
+1 
Is there any update on this yet?

**Comment by jerielng** [comment]
Hey @thainx, we are working on a solution and hope to have this out in the next few weeks after some testing. In the meantime, as a workaround, are you able to work off of a fork using #241 as a reference?

**Comment by AbhavThakur** [comment]
Hi @jerielng 
I am able to fix it by removing  compilerOptions from node_modules in braze and now working fine.

**Comment by cam-shaw** [comment]
Here's a patch file for anyone that needs it

https://gist.github.com/cam-shaw/0bf9a0a72aedc39c8be0e2fe9a50d315

**Comment by thainx** [comment]
Hi @jerielng , I was able to compile by removing the <source/target>Compatibility and jvmTarget (with `patch-package`). Hopefully it won't have any runtime issue.
Looking forward to an official update from Braze 😀 
Thanks for your kind support!

**Comment by AbhavThakur** [comment]
Hi @jerielng 
When can we expect the official braze version .
Can you provide a tentative date .
So, our team can plan the release.

**Comment by jerielng** [comment]
Hey @AbhavThakur, our team is currently working to review and test the upcoming release. Barring any unexpected circumstances, we hope to have this out sometime in the next two weeks or so.

**Comment by vanessaland** [comment]
Hey everyone, we've just [released version 8.3.0](https://github.com/braze-inc/braze-react-native-sdk/releases/tag/8.3.0) of the Braze React Native SDK! This should now resolve the compatibility issue with 0.73. 

I'm going to close the issue for now, but feel free to reopen it if any further issues arise!
